src/dotnet/CarbonAware.Aggregators/src/CarbonAware/CarbonAwareAggregator.cs

```diff
@@ -2,28 +2,126 @@
+    /// </summary>
+    /// <param name="logger">The logger for the aggregator</param>
+    /// <param name="dataSource">An <see cref="ICarbonIntensityDataSource"> data source.</param>
     public CarbonAwareAggregator(ILogger<CarbonAwareAggregator> logger, ICarbonIntensityDataSource dataSource)
     {
         this._logger = logger ?? throw new ArgumentNullException(nameof(logger));
         this._dataSource = dataSource;
     }
 
+    /// <inheritdoc />
     public async Task<IEnumerable<EmissionsData>> GetEmissionsDataAsync(IDictionary props)
     {
-        DateTimeOffset end = GetOffsetOrDefault(props, CarbonAwareConstants.End, DateTimeOffset.Now);
-        DateTimeOffset start = GetOffsetOrDefault(props, CarbonAwareConstants.Start, end.AddDays(-7));
-        _logger.LogInformation("Aggregator getting carbon intensity from data source");
-        return await this._dataSource.GetCarbonIntensityAsync(GetLocationOrThrow(props), start, end);
+        using (var activity = Activity.StartActivity())
+        {
+            DateTimeOffset end = GetOffsetOrDefault(props, CarbonAwareConstants.End, DateTimeOffset.Now);
+            DateTimeOffset start = GetOffsetOrDefault(props, CarbonAwareConstants.Start, end.AddDays(-7));
+            _logger.LogInformation("Aggregator getting carbon intensity from data source");
+            return await this._dataSource.GetCarbonIntensityAsync(GetLocationOrThrow(props), start, end);
+        }
+    }
+
+    /// <inheritdoc />
+    public async Task<IEnumerable<EmissionsForecast>> GetCurrentForecastDataAsync(IDictionary props)
+    {
+        using (var activity = Activity.StartActivity())
+        {
+            DateTimeOffset start = GetOffsetOrDefault(props, CarbonAwareConstants.Start, DateTimeOffset.Now);
+            DateTimeOffset end = GetOffsetOrDefault(props, CarbonAwareConstants.End,  start.AddDays(1));
+            _logger.LogInformation("Aggregator getting carbon intensity forecast from data source");
+
+            var forecasts = new List<EmissionsForecast>();
+            foreach(var location in GetLocationOrThrow(props))
+            {
+                var forecast = await this._dataSource.GetCurrentCarbonIntensityForecastAsync(location);
+                forecast.StartTime = start;
+                forecast.EndTime = end;
+                forecast.ForecastData = FilterByDate(forecast.ForecastData, start, end);
+                if(forecast.ForecastData.Any())
+                {
+                    TransformRollingWindow(forecast, props);
+                    forecast.OptimalDataPoint = GetOptimalEmissions(forecast.ForecastData);
+                }
+                forecasts.Add(forecast);
+            }
+
+            return forecasts;
+        }
+    }
+
+    private IEnumerable<EmissionsData> FilterByDate(IEnumerable<EmissionsData> data, DateTimeOffset start, DateTimeOffset end)
+    {
+        return data.Where(ed => ed.Time >= start && ed.Time < end);
```

Comment:
It can be replaced with `data.Where(ed => ed.TimeBetween(startDate.DateTime, end.DateTime))`. There are other parts of the code base that follows this call.

Reply:
I was using that originally and my unit tests were failing because `TimeBetween` excludes matching `startTime` which I did not want.  Not knowing what else relied on the `exclusive start / inclusive end` for business logic I opted not to touch it